storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/BytesColumnVector.java

```diff
@@ -49,13 +49,14 @@
    */
   public int[] length;
 
-  // A call to increaseBufferSpace() or ensureValPreallocated() will ensure that buffer[] points to
-  // a byte[] with sufficient space for the specified size.
-  private byte[] buffer;   // optional buffer to use when actually copying in data
+  // Calls to ensureValPreallocated() ensure that buffer starting at nextFree
+  // points to a byte[] with sufficient space for the specified size.
```

Comment:
I find the use of smallBuffer and buffer rather confusing after refactoring.
Would it make sense to rename to something more descriprive like currBuffer and next Buffer?

Reply:
Yes, I agree. How about currentValue/currentOffset and sharedBuffer/sharedBufferOffset?

Reply:
The new names are meaningful